src/SFA.DAS.Roatp.CourseManagement.Jobs/Infrastructure/ApiClients/ApiClientBase.cs

```diff
@@ -0,0 +1,153 @@
+                    new StringContent(serializeObject, Encoding.UTF8, _contentType)))
+                {
+                    await LogErrorIfUnsuccessfulResponse(response);
+                    return response.StatusCode;
+                }
+            }
+            catch (HttpRequestException ex)
+            {
+                _logger.LogError(ex, $"Error when processing request: {HttpMethod.Post} - {uri}");
+                throw;
+            }
+        }
+
+        /// <summary>
+        /// HTTP POST to the specified URI
+        /// </summary>
+        /// <typeparam name="T">The type of the object to POST.</typeparam>
+        /// <typeparam name="U">The type of the object to read.</typeparam>
+        /// <param name="uri">The URI to the end point you wish to interact with.</param>
+        /// <returns>A Task yielding the result (of type U).</returns>
+        /// <exception cref="HttpRequestException">Thrown if something unexpected occurred when sending the request.</exception>
+        protected async Task<U> Post<T, U>(string uri, T model)
+        {
+            var serializeObject = JsonConvert.SerializeObject(model);
+
+            try
+            {
+                using (var response = await _httpClient.PostAsync(new Uri(uri, UriKind.Relative),
+                    new StringContent(serializeObject, Encoding.UTF8, _contentType)))
+                {
+                    await LogErrorIfUnsuccessfulResponse(response);
+                    return await response.Content.ReadAsAsync<U>();
+                }
+            }
+            catch (HttpRequestException ex)
+            {
+                _logger.LogError(ex, $"Error when processing request: {HttpMethod.Post} - {uri}");
+                throw;
+            }
+        }
+
+        private async Task LogErrorIfUnsuccessfulResponse(HttpResponseMessage response)
+        {
+            if (response?.RequestMessage != null && !response.IsSuccessStatusCode)
+            {
+                var callingMethod = new System.Diagnostics.StackFrame(1).GetMethod().Name;
+
+                var httpMethod = response.RequestMessage.Method.ToString();
+                var statusCode = (int)response.StatusCode;
+                var reasonPhrase = response.ReasonPhrase;
+                var requestUri = response.RequestMessage.RequestUri;
+
+                var responseContent = await response.Content.ReadAsStringAsync();
+                var apiErrorMessage = responseContent;
+
+                if (TryParseJson<ApiError>(responseContent, out var apiError) && !string.IsNullOrWhiteSpace(apiError?.Message))
+                {
+                    apiErrorMessage = apiError.Message;
+                }
+
+                _logger.LogError($"Method: {callingMethod} || HTTP {statusCode} {reasonPhrase} || {httpMethod}: {requestUri} || Message: {apiErrorMessage}");
+            }
+        }
+
+        private static bool TryParseJson<T>(string json, out T result)
```

Comment:
We are calling our inner apis, if they are not returning any json content in case of error then this does not make sense. Or we should make our inner api return json content to make sense of this.

Reply:
I'll remove the attempt to parse json

Reply:
Is it possible this is there to deal with WAF generated notifications? I've taken it out already, but let's discuss